tests/devices/qubit/test_initialize_state.py

```diff
@@ -58,28 +58,24 @@ def test_create_initial_state_with_BasisState(self):
         state[0, 1, 0] = 0  # set to zero to make test below simple
         assert qml.math.allequal(state, np.zeros((2, 2, 2)))
 
-    def test_create_initial_state_with_StatePrep(self):
+    @pytest.mark.parametrize("prep_op_cls", [qml.StatePrep, qml.AmplitudeEmbedding])
```

Comment:
Should we add all subclasses of StatePrepBase here?

Reply:
I think this is for `StatePrep` specifically, I just added this because it's effectively a `StatePrep` lol